sphinx/source/docs/user_guide/examples/interaction_checkbox_button_group.py

```diff
@@ -4,8 +4,8 @@
 LABELS = ["Option 1", "Option 2", "Option 3"]
 
 checkbox_button_group = CheckboxButtonGroup(labels=LABELS, active=[0, 1])
-checkbox_button_group.js_on_click(CustomJS(code="""
-    console.log('checkbox_button_group: active=' + this.active, this.toString())
+checkbox_button_group.js_on_event("button_click", CustomJS(code="""
+    console.log('checkbox_button_group: active=' + this.origin.active, this.toString())
```

Comment:
Sounds good, will make that update in the examples/tests.

Reply:
It's probably not as clear in the docs as it needs to be but `cb_obj` means different things for property change (`js_on_change`) callbacks than it does for event (`js_on_event`) callbacks. For property change callbacks, it contains the object that has the property the callback was on ("callback object"). That was thecase for nearly a decade, long before event callbacks existed. When event callbacks were added, we wanted to be able to have users re-use the same `CustomJS` objects. That meant we had to pass _something_ as `cb_obj` to the JS code. But it does not make sense to pass the "callback object" and in fact sometimes there is not one at all, e.g. for `Document` callbacks. But there are "event objects" that contain event-specific information that callbacks might need. That is what is passed for `cb_obj` for event callbacks. 

You can see all of them here:

https://docs.bokeh.org/en/latest/docs/reference/events.html

Some of them (e.g. button click) are empty, but most of them are not. Button click is empty because there is no further information to pass along, the information is simply that a button click occurred and the event itself conveys that. But in any case, they are not useless in general, because in general they contain more information. However, this is also why for event callbacks, if references to models are needed, we (or at least I) recommend using the `args` dict. Also this really on affects the "stateful buttons".  Options could be to make new different "stateful button click" events but that starts to get overly complex for users IMO, or as @mattpap suggests, just not have  these be button click emitters at all. 

TLDR; historical quirks in an old OSS project

Reply:
I think we're good here.